src/integration/java/io/pinecone/integration/controlPlane/pod/CreateDescribeListAndDeleteIndexTest.java

```diff
@@ -13,37 +13,28 @@
 import static org.junit.jupiter.api.Assertions.*;
 
 public class CreateDescribeListAndDeleteIndexTest {
-    private static final String indexName = RandomStringBuilder.build("create-index", 8);
-    private static final String indexPodType = "p1.x1";
-    private static Pinecone controlPlaneClient = new Pinecone.Builder(System.getenv("PINECONE_API_KEY")).build();
+
+    private static final IndexManagerSingleton indexManager = IndexManagerSingleton.getInstance();
+    private static Pinecone controlPlaneClient = indexManager.getPineconeClient();
+    private static String indexName;
+    private static int indexDimension;
+    private static String indexPodType;
     private static final String environment = System.getenv("PINECONE_ENVIRONMENT");
 
     @BeforeAll
     public static void setUp() throws InterruptedException {
-        // Create the index
-        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType(indexPodType);
-        CreateIndexRequestSpec createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
-        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
-                .name(indexName)
-                .metric(IndexMetric.COSINE)
-                .dimension(10)
-                .spec(createIndexRequestSpec);
-        controlPlaneClient.createIndex(createIndexRequest);
-        waitUntilIndexIsReady(controlPlaneClient, indexName);
-    }
-
-    @AfterAll
-    public static void cleanUp() {
-        // Delete the index
-        controlPlaneClient.deleteIndex(indexName);
+        indexName = indexManager.getPodIndexName();
+        indexDimension = indexManager.getDimension();
+        IndexModel podIndex = indexManager.getPodIndexModel();
+        indexPodType = podIndex.getSpec().getPod().getPodType();
```

Comment:
LOVE how much code is being removed from individual tests

Reply:
Yeah this is so sweet